integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/search/predicate/QueryStringPredicateSpecificsIT.java

```diff
@@ -0,0 +1,275 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.integrationtest.backend.tck.search.predicate;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.hibernate.search.util.impl.integrationtest.common.assertion.SearchResultAssert.assertThatQuery;
+
+import java.util.List;
+import java.util.function.Supplier;
+
+import org.hibernate.search.engine.backend.common.DocumentReference;
+import org.hibernate.search.engine.search.common.Rewrite;
+import org.hibernate.search.engine.search.predicate.SearchPredicate;
+import org.hibernate.search.engine.search.predicate.dsl.QueryStringPredicateFieldStep;
+import org.hibernate.search.engine.search.predicate.dsl.QueryStringPredicateOptionsStep;
+import org.hibernate.search.engine.search.predicate.dsl.SearchPredicateFactory;
+import org.hibernate.search.engine.search.query.SearchQuery;
+import org.hibernate.search.util.common.SearchException;
+import org.hibernate.search.util.impl.integrationtest.mapper.stub.StubMappingScope;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+class QueryStringPredicateSpecificsIT extends AbstractBaseQueryStringPredicateSpecificsIT<QueryStringPredicateFieldStep<?>> {
+
+	@ParameterizedTest
+	@MethodSource("rewriteOptions")
+	void rewriteOptions(Rewrite rewrite, Integer n) {
+		StubMappingScope scope = index.createScope();
+		String absoluteFieldPath = index.binding().analyzedStringField1.relativeFieldName;
+
+		QueryStringPredicateOptionsStep<?> optionsStep = scope.predicate().queryString()
+				.field( absoluteFieldPath )
+				.matching( TERM_1 );
+
+		SearchPredicate predicate = ( n == null ? optionsStep.rewrite( rewrite ) : optionsStep.rewrite( rewrite, n ) )
+				.toPredicate();
+
+		SearchQuery<DocumentReference> query = scope.query()
+				.where( predicate )
+				.toQuery();
+		assertThatQuery( query )
+				.hasDocRefHitsAnyOrder( index.typeName(), DOCUMENT_1, DOCUMENT_2 );
```

Comment:
If the query string and search results are always the same, does this really test the rewrite options, or just that none blows up? :thinking:

Reply:
> or just that none blows up

just that, I haven't figured out yet how to make a test where a rewrite is making a "visible difference" 😃